### PR TITLE
Milestone user interface

### PR DESCRIPTION
**Pull request by MiguelDP4**
- Created a game board array
- Created a method to print the game board
- Created mock game logic (intended to be transferred to a library in the next step)

### CODE REVIEW

File: bin/main [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  puts "Player #{player_turn}, select which cell you want to fill by writing it's cell number"
+  invalid_input = true
+  while invalid_input
+    number = gets.chomp
+    number = number.to_i
+    invalid_input = number < 1 || number > 9
+    puts 'Please only write a number from 1 to 9' if invalid_input
+    if board_array[number] == " "
+      board_array[number] = "X" if player_turn == 1
+      board_array[number] = "O" if player_turn == 2
```
Comment (stratospherique) [inline bin/main]: - **DRY** your code 

```suggestion
      board_array[number] = player_turn == 1 ? 'X' : 'Y'
```

File: bin/main [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      puts "Only play in an empty cell"
+      invalid_input = true
+    end
+  end
+  
+  if player_turn == 1
+    player_turn = 2
+  else
+    player_turn = 1
+  end
```
Comment (stratospherique) [inline bin/main]:
```suggestion
    player_turn = player_turn == 1 ? 2 : 1
```

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work @team :clap: :clap: However, some requirements need to be implemented in order for us to approve your project. Check :arrow_down: 

##### Behaviour

* [ ] In this step not everything must work, the point is to make sure that students thought about everything listed below. They might do it only with inline comments. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [ ] Basic game flow is prepared: ([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
  * [x] `[REQUIRED]` game tells which player turn it is
  * [x] `[REQUIRED]` game informs player if selected move is invalid
  * [ ] `[REQUIRED]` game informs player if selected move is a winning move
  * [ ] `[REQUIRED]` game informs player if selected move is a draw move
  
- You need to implement or just write in comments the logic related to verifying if a player move should halt the game (player wins or draw) or not.

### That's it :relieved: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
_**Ahmed Mahfoudh**_

**Review by Forison — CHANGES_REQUESTED** [review]
## Status: Require Changes🛑
# Hi team🙋‍♂️
Good job on implementing some of the changes raised in the previous review.
There are a few behavioral flaws that we have to address in this particular milestone.
* [x] game informs player if the selected move is a winning move.([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] game informs player if the selected move is a draw move ([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x]  The game ends currently on the eighth moves. When the ninth move is the decider for a win or draws the game does not respond to accordingly. It is as if the ninth move is not taking into consideration by the game.
## That is it, kindly make this change and submit the project for review🙂
### Happy coding💻
### Addo Forison

**Review by fatymahmed — APPROVED** [review]
### Hi @MiguelDP4 @bren2102 ,
### Great job making the suggested changes🎉🎉
[OPTIONAL]
- Your main file should be `main.rb`
- Please do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from Pathwright. Use [these](https://github.com/microverseinc/linters-config/tree/master/ruby) config files.
### It's a 👍🏾 from me!
## ✔️ Approved!

Good luck with your next projects.
 _Don't forget to submit your project completion form_

**Pull request merged**